src/io/metadata.cpp

```diff
@@ -214,6 +217,13 @@ void Metadata::CheckOrPartition(data_size_t num_all_data, const std::vector<data
       Log::Fatal("Weights size doesn't match data size");
     }
 
+    // check positions
+    if (!positions_.empty() && num_positions_ != num_data_) {
+      positions_.clear();
+      num_positions_ = 0;
+      Log::Fatal("Positions size doesn't match data size");
```

Comment:
```suggestion
      Log::Fatal("Positions size (%i) doesn't match data size (%i)", num_positions_, num_data_);
```

Since you already have access to them, please print the sizes here, to help users who encounter this error debug what went wrong.